### ISSUE TITLE
[ISSUE] Issue with `databricks_system_schema` resource 

### ISSUE THREAD

**Issue by Waiwait** [databricks/terraform-provider-databricks#4066]
### Configuration

`databricks_system_schema` resource passes silently when a schema has been disabled outside of terraform. This was discovered when we switched the metastore for a workspace and then dependent grants on the `system_schema` started failing after not being able to access the table. 

### Expected Behavior
Terraform correctly recognises that the schema has been disabled outside of the state and acts appropriately.

### Actual Behavior
Terraform silently passes, assuming that the schema is still enabled.

### Terraform and provider versions

`terraform-provider-databricks=1.49.0`
`terraform=v1.55.0`

### Would you like to implement a fix?

I think this is caused by the `Read` function simply returning `nil` at the end of the for loop if no appropriate schema is found. This would be fixed by adding `d.SetId("")` to this, so that on the next run, it'll force a `Create` call. Happy to implement, just wanted to make sure this was the right approach.

**Comment by alexott** [comment]
yes, we're using similar approach in other resources, like, library: https://github.com/databricks/terraform-provider-databricks/blob/main/clusters/resource_library.go#L86

**Comment by alexott** [comment]
Fixed by #4068
